src/app/states/firebase.service.ts

```diff
@@ -40,25 +52,30 @@ export class FirebaseService {
   private configAppIdKey = 'firebase-app-id';
   private configProjectIdKey = 'firebase-project-id';
   private emailKey = 'firebase-email';
-  private passwordKey = 'firebase-password';
 
-  private onLoad = true;
+  private costCollection = 'cost';
+  private totalsPath = 'info/totals';
+  private configPath = 'info/config';
+  private conversationCollection = 'conversation';
 
-  loginState = new BehaviorSubject<LoginState>('out');
+  loginState = new BehaviorSubject<LoginState>('load');
   error = new BehaviorSubject<string>('');
+  password = new BehaviorSubject<string>('');
+  private uuid: string|null = null;
```

Comment:
these private variables create some side effects. it would reduce issues if they are part of the state model. if you want i can have a look at after merge.

Reply:
If you could do that that would be great. I think showing a better pattern will be easier than trying to explain it to me.